src/helper_functions.cpp

```diff
@@ -53,8 +39,11 @@ std::string readStringWcharTag(const BasicIo::UniquePtr& io, size_t length) {
   Internal::enforce(length <= io->size() - io->tell(), Exiv2::ErrorCode::kerCorruptedMetadata);
   DataBuf FieldBuf(length + 1);
   io->readOrThrow(FieldBuf.data(), length, ErrorCode::kerFailedToReadImageData);
-  std::u16string wst(FieldBuf.begin(), FieldBuf.end());
-  return utf16ToUtf8(wst);
+  std::string wst(FieldBuf.begin(), FieldBuf.end() - 1);
```

Comment:
AFAIK this will break UTF-8 conversion. Unfortunately, I'm away from my computer for ~2 months so I can't test properly. I don't think the current tests have UTF8 data.

Reply:
Is this fixed now? I haven't been following what's been happening, but I see that all the checks are passing and the code changes look good to me, so I'm happy to approve this if everything's working.

Reply:
Might as well merge. If it's not working properly with UTF8 data, it's at least better than before.